packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -222,9 +169,15 @@ - (void)finishTransaction:(FlutterMethodCall *)call result:(FlutterResult)result
   SKPaymentTransaction *transaction =
       [self.paymentQueueHandler.transactions objectForKey:identifier];
   if (!transaction) {
-    result([FlutterError errorWithCode:@"storekit_platform_invalid_transaction"
-                               message:@"Invalid transaction ID is used."
-                               details:call.arguments]);
+    result([FlutterError
+        errorWithCode:@"storekit_platform_invalid_transaction"
+              message:[NSString
+                          stringWithFormat:@"The transaction with transactionIdentifer:%@ does not "
+                                           @"exsit. Note that if the transactionState is "
+                                           @"purchasing, the transactionIdentifier will be "
+                                           @"nil(null). And you should not finish this transaction",
```

Comment:
> And you should not finish this transaction

Can you go into more detail about this?

Reply:
Sure, by apple's document finish transaction with transactionIdentifier of purchasing will throw exception; I will put something similar here.